app/Models/Album.php

```diff
@@ -268,4 +274,38 @@ public function newEloquentBuilder($query): AlbumBuilder
 	{
 		return new AlbumBuilder($query);
 	}
+
+	/**
+	 * Set the GPX track for the album.
+	 *
+	 * @param UploadedFile $file the GPX track file to be set
+	 *
+	 * @return void
+	 */
+	public function setTrack(UploadedFile $file): void
+	{
+		if ($this->track_short_path != null) {
+			Storage::delete($this->track_short_path);
+		}
+
+		$new_track_id = sha1($file);
```

Comment:
```suggestion
		$new_track_id = hash('sha256',$file->get());
```
I am not entirely sure of this but this should avoid using sha1 and this should make sure we are hashing the content of the file.

Reply:
Fixed (but my commit message was wrong, should be SHA256)

Reply:
What if somebody uploads the same track file to two different albums? That's not even that unlikely, to be honest. We need to make sure that we handle it well...